Support for HT-A7000

Flameeyes opened the pull request:
This builds on top of the other pull request I sent to fix typing, to include some additional detail in structures, and provide better access to the content change structure.

Flameeyes committed:
Include serialNumber, generation, and model in Sysinfo.

These fields are present on HT-A7000 soundbars.

Flameeyes committed:
Make ContentChange include the full PlayInfo structure.

The structure of content change should include all of the attributes
from PlayInfo with the same schema. This allows actually receiving all
of the information for PlayInfo as notifications.

Flameeyes committed:
Include the thumbnail URL if provided by the device.

This in particular is provided by HT-A7000.

rytilahti commented:
Could you please rebase this on top of the current master? Looks otherwise good to me, but I cannot test this personally right now.

rytilahti commented on songpal/notification.py:
```diff
@@ -153,23 +159,11 @@ def __str__(self):
-    source = attr.ib()
-    output = attr.ib()
-    uri = attr.ib()
-    applicationName = attr.ib()
```
Not all of these seem to be available in `PlayInfo`, at least `applicationName` seems to be missing.

Flameeyes replied:
Turns out `applicationName` was the only one missing and it was meant to exist in `PlayInfo` as well.

Added as a pre-requisite commit.

Flameeyes committed:
Add missing `applicationName` field to `PlayInfo`.

This is documented when `source` is `cast:audio` for both
`getPlayingContentInfo` and `notifyPlayingContentInfo`.

Flameeyes committed:
Make `SoftwareUpdateChange` consistent with `SoftwareUpdateInfo`.

This addresses #93 by separating the `swInfo` attribute into a separate
class `SoftwareUpdateInfoDetails`, and making `SoftwareUpdateChange` a
simpler derivation off two other classes.

This follows the API docs for `getSWUpdateInfo` and `notifySWUpdateInfo`
available at https://developer.sony.com/develop/audio-control-api/documentation/api-reference#_getswupdateinfo_v1_0

rytilahti reviewed (approved):
LGTM, thanks! :+1:
